datadog_checks_base/datadog_checks/base/checks/openmetrics/v2/base.py

```diff
@@ -54,6 +57,13 @@ def __init__(self, name, init_config, instances):
         self.check_initializations.append(self.configure_scrapers)
 
     def check(self, _):
+        """
+        Perform an openmetrics-based check.
+
+        Subclasses should not need to override this, add custom scrapers instead.
```

Comment:
I'm uncertain that this is necessarily true or expected to be true.

After a quick search, there's at least one instance in which this is overridden: https://github.com/DataDog/integrations-core/blob/e31b6042cf404773142bc673ffcf9d6f7449fb1a/avi_vantage/datadog_checks/avi_vantage/check.py#L95-L101

Admittedly, that looks like something that might be possible to rewrite to make use of scrapers.

More generally, I wonder if there's a use case for an integration to collect metrics both openmetrics metrics as well as other metrics (without needing to define two check instances) 🤔 .

Reply:
Good point, thanks for searching! Do you think that *should* covers this? Even with your example, I get the impression that `check` contains important logic that we don't want to override but rather extend.